### ISSUE TITLE
usethis::use_pkdgdown_github_pages error with surprising error message

### ISSUE THREAD

**Issue by mikeroswell** [r-lib/usethis#1470]
I saw a previous issue was closed": _Originally posted by @jennybc in https://github.com/r-lib/usethis/issues/1447#issuecomment-839849466_

But I'm running into the same basic error but it's less obvious to me that this is on the GH side and not in `usethis` or `pkgdown`... I wonder if there might be a typo somewhere with "ed" instead of "red":
```
Error in gh_process_response(raw) : 
GitHub API error (500): 
Message: 
Type .Last.error.trace to see where the error occured
> .Last.error.trace

 Stack trace:

Error: 'col_ed' is not an exported object from 'namespace:cli'
> 
```

**Comment by mikeroswell** [comment]
I get the same error here

```
> use_github_pages()
✓ Initializing empty, orphan 'gh-pages' branch in GitHub repo 'mikeroswell/MeanRarity'
Error in gh_process_response(raw) : 
GitHub API error (500): 
Message: 
Type .Last.error.trace to see where the error occured
> .Last.error.trace

 Stack trace:

Error: 'col_ed' is not an exported object from 'namespace:cli'
```

**Comment by jennybc** [comment]
I think you really are getting a 500 error from GitHub and then a cli error while attempting to *show* you more about the error. Off the top of my head, I'm guessing you need to update cli and that someone has failed to bump a minimum version, probably gh.

**Comment by mikeroswell** [comment]
Nice! Re-installing `cli` from CRAN seemed to work, and now I get this: 


```
Error in gh_process_response(raw) : 
GitHub API error (500): 
Message: 
Type .Last.error.trace to see where the error occurred
> .Last.error.trace
$calls
$calls[[1]]
use_pkgdown_github_pages()

$calls[[2]]
use_github_pages()

$calls[[3]]
create_gh_pages_branch(tr, branch = "gh-pages")

$calls[[4]]
gh("POST /repos/{owner}/{repo}/git/commits", message = "first commit", 
    tree = sha_empty_tree)

$calls[[5]]
gh::gh(endpoint, ..., owner = tr$repo_owner, repo = tr$repo_name, 
    .api_url = tr$api_url)

$calls[[6]]
gh_process_response(raw)

$calls[[7]]
gh_error(response)

$calls[[8]]
throw(cond)


$parents
[1] 0 1 2 3 4 5 6 7

$envs
$envs[[1]]
[1] "0x7fac039e6c70"

$envs[[2]]
[1] "0x7fac07870c80"

$envs[[3]]
[1] "0x7fac01e6b648"

$envs[[4]]
[1] "0x7fac01c1b350"

$envs[[5]]
[1] "0x7fac01c192a8"

$envs[[6]]
[1] "0x7fac05bdffa0"

$envs[[7]]
[1] "0x7fac05bdefa8"

$envs[[8]]
[1] "0x7fac05c02ad8"


$topenvs
$topenvs[[1]]
[1] "namespace:usethis"

$topenvs[[2]]
[1] "namespace:usethis"

$topenvs[[3]]
[1] "namespace:usethis"

$topenvs[[4]]
[1] "namespace:usethis"

$topenvs[[5]]
[1] "namespace:gh"

$topenvs[[6]]
[1] "namespace:gh"

$topenvs[[7]]
[1] "namespace:gh"

$topenvs[[8]]
[1] "0x7fac066e1070"


$indices
[1] 1 2 3 4 5 6 7 8

$nframes
[1] 8

$messages
$messages[[1]]
[1] "\nGitHub API error (500): \nMessage: "


$ignore
NULL

$classes
[1] "github_error"   "http_error_500" "error"          "condition"     

$pids
[1] 32427 32427 32427 32427 32427 32427 32427 32427

attr(,"class")
[1] "rlib_trace"
```

**Comment by jennybc** [comment]
Hello @gaborcsardi do you think current gh assumes a higher version of cli than it currently has in DESCRIPTION (which is `cli (>= 2.0.1)`)?

**Comment by gaborcsardi** [comment]
Possibly.

**Comment by gaborcsardi** [comment]
The `col_ed` thing is (was) not an error in cli, it is (was) an error in gh, fixed in version 1.3.0:
https://github.com/cran/gh/commit/2dcdddb8b34bafa0d27cdb896f6346b114c1cd04#diff-d0b4ccfee35b04daaa283121902133ab90336582d08b4001dd848095c03c02d0L726
